include/psa/crypto_values.h

```diff
@@ -2011,4 +2097,154 @@ static inline int mbedtls_svc_key_id_is_null( mbedtls_svc_key_id_t key )
+
+/** The second peer in a balanced PAKE.
+ *
+ * Although balanced PAKE algorithms are symmetric, some of them needs an
+ * ordering of peers for the transcript calculations. If the algorithm does not
+ * need this, either #PSA_PAKE_SIDE_FIRST or #PSA_PAKE_SIDE_SECOND are
+ * accepted.
+ */
+#define PSA_PAKE_SIDE_SECOND                ((psa_pake_side_t)0x02)
+
+/** The client in an augmented PAKE.
+ *
+ * Augmented PAKE algorithms need to differentiate between client and server.
+ */
+#define PSA_PAKE_SIDE_CLIENT                ((psa_pake_side_t)0x11)
+
+/** The server in an augmented PAKE.
+ *
+ * Augmented PAKE algorithms need to differentiate between client and server.
+ */
+#define PSA_PAKE_SIDE_SERVER                ((psa_pake_side_t)0x12)
+
+/** The PAKE uses elliptic curves.
+ *
+ * The corresponding family type is ::psa_ecc_family_t. In determining a
+ * specific curve in the family the cipher suite (see
+ * ::psa_pake_cipher_suite_t) bits are interpreted in the exact same way
+ * as key bits are.
+ *
+ * Input and output during the operation can involve group elements and scalar
+ * values:
+ * -# The format for group elements is the same as for public keys on the
+ *  specific curve would be. For more information, consult the documentation of
+ *  psa_export_public_key().
+ * -# The format for scalars is the same as for private keys on the specific
+ *  curve would be. For more information, consult the documentation of
+ *  psa_export_key().
+ */
+#define PSA_PAKE_PRIMITIVE_TYPE_ECC       ((psa_pake_primitive_type_t)0x01)
+
+/** The PAKE uses finite fields based Diffie-Hellman groups.
+ *
+ * The corresponding family type is ::psa_dh_family_t. In determining a
+ * specific group in the family the cipher suite (see
+ * ::psa_pake_cipher_suite_t) bits are interpreted in the exact same way
+ * as key bits are.
+ *
+ * Input and output during the operation can involve group elements and scalar
+ * values:
+ * -# The format for group elements is the same as for public keys on the
+ *  specific group would be. For more information, consult the documentation of
+ *  psa_export_public_key().
+ * -# The format for scalars is the same as for private keys on the specific
+ *  group would be. For more information, consult the documentation of
+ *  psa_export_key().
+ */
+#define PSA_PAKE_PRIMITIVE_TYPE_DH       ((psa_pake_primitive_type_t)0x02)
+
+/** The key share being sent to or received from the peer.
+ *
+ * Unless the documentation of the PAKE algorithm says otherwise this is a
+ * group element.
+ *
+ * For information regarding representation consult the documentation of
+ * individual ::psa_pake_primitive_type_t constants.
+ *
+ * Some PAKE algorithms need to exchange several key shares. If that is the
+ * case, this value marks the first key share sent and the first key share
+ * received. For values sent or received afterwards, use
+ * #PSA_PAKE_DATA_KEY_SHARE_2 and #PSA_PAKE_DATA_KEY_SHARE_3.
```

Comment:
Do we need `PSA_PAKE_DATA_KEY_SHARE_2` and `PSA_PAKE_DATA_KEY_SHARE_3`, or can we say that the first share is the first call to `psa_pake_input with `PSA_PAKE_DATA_KEY_SHARE`, the second share is the second call, etc.?

Reply:
I have to agree that this would be neat.

Presumably, PAKE constructions that have multiple key shares will always consume and produce the key shares in numerical order? - so using `PSA_PAKE_DATA_SHARE_X` would demand step-sequence validation, and provides no functional utility over @gilles-peskine-arm's suggestion of a single data type/step.